Core/include/Acts/Material/interface/IAssignmentFinder.hpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/Material/MaterialInteraction.hpp"
+
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#pragma once
+
+namespace Acts {
+
+/// @brief Interface for the material mapping that seeks the possible
+/// assignment candidates for the material interactiosn
+class IAssignmentFinder {
+ public:
+  /// Virtual destructor
+  virtual ~IAssignmentFinder() = default;
+
+  /// @brief SurfaceAssignment is a surface, a position and a direction
+  using SurfaceAssignment = std::tuple<const Surface*, Vector3, Vector3>;
+
+  /// @brief VolumeAssignment is a volume and a entry and exit of the volume
+  using VolumeAssignment =
+      std::tuple<const InteractionVolume, Vector3, Vector3>;
```

Comment:
where is `InteractionVolume` defined? should this be `const *` or can `const` be dropped?

same there: a struct might improve readability

Reply:
`const` can be dropped.

Reply:
This will go after we harmonise Gen1 + Gen2 geometry into Gen3.